test/scopes_test.js

```diff
@@ -99,3 +100,98 @@ suite('scopeMatch', function() {
     mktest(['foo:bar'], [], false));
 });
 
+suite('scopeIntersection', () => {
```

Comment:
It would be good to check that commutativity holds for all of these cases.  Maybe a general `testScopeIntersection(scope1, scope2, expected)` that tested that scope1 ∩ scope2 == scope2 ∩ scope 1 == expected?

Reply:
I like that idea, thanks for suggesting it.